aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,65 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  // init to column major format
+  // Note: `sol` and `lu` need to be contiguous since we pass
+  // sol.data() and lu.data() to Lapack
+  if (&self == &sol) {
+    // eg. torch.gesv(b, A, out=(b, A))
+    if (isTransposeContiguous(sol)) {
+      sol.t_();
+    } else {
+      auto Bc = self.view({bx, by}).t().clone();
+      sol.resize_({by, bx}).copy_(Bc);
+    }
+  } else if (sol.numel() == self.numel() &&
+             isTransposeContiguous(sol)){
+    // allow reuse
+    sol.t_().copy_(self.view({bx, by}).t());
```

Comment:
this breaks 
1. if `self` is not viewable to `{bx, by}`, or
2. `sol` doesn't have correct shape (but has correct `numel`)

Reply:
1. since `bx` and `by` are dimensions of `self`, it should always be viewable as `{bx, by}` (this is done to simplify the code for handling both 1D and 2D `b` Tensors.
2. Fixed in the latest commit to always call resize to the target output size regardless of contiguity or `numel`